doc/source/docs/user_guide/sampling_and_grids.ipynb

```diff
@@ -0,0 +1,518 @@
+   "metadata": {},
+   "source": [
+    "For a square grid, points are arranged in squares. By default, grids are also clipped to the extent of the shapes being gridded:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "30557c89-47ad-475c-a638-8694ce6edd81",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "m = nybb.explore()\n",
+    "squaregrid_cover.explore(m=m, color='red')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9734ae2e-d882-43ad-8f8d-d942809dac14",
+   "metadata": {},
+   "source": [
+    "For a \"hexagonal\" grid, points are arranged in hexagons (or triangles, depending on your perspective): "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9245b14b-ce1d-45eb-bbfe-95b10f2b0c6c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "hexgrid_cover = geopandas.make_grid(nybb, size=30, tile='hex')\n",
+    "m = nybb.explore()\n",
+    "hexgrid_cover.explore(color='red', m=m)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "23b6757b",
+   "metadata": {},
+   "source": [
+    "You can also specify how far apart points should be using the `spacing` option, and keep all of the generated points within the GeoDataFrame's bounding box by setting the `clip` option to False:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f5c62604",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "hexgrid_cover = geopandas.make_grid(nybb, spacing=10000, tile='hex', clip=False)\n",
+    "m = nybb.explore()\n",
+    "hexgrid_cover.explore(color='red', m=m)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0e052d6a-aabd-431c-8ef1-0c3d7d3822eb",
+   "metadata": {},
+   "source": [
+    "To sample a grid like this from *within* a shape, you can use the `.sample_points()` method with the option `method='grid'`. This grids *each* shape separately, and supports similar options to `geopandas.grids.make_grid()` above."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e6e54965-b474-4cb9-a1e6-77956b458845",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "m = nybb.explore()\n",
+    "nybb.sample_points(method='grid', size=10).explore(m=m, color='red')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "05f4a760-92c0-44dc-bfa4-43383cd5a248",
+   "metadata": {},
+   "source": [
+    "Note how the densities of the grid varies by shape in this case. This is because the `sample_points()` methods will sample the same \"size\" of grid for each shape. For example, each shape below is sampling a ten-by-ten hexagonal grid, so smaller shapes will have more densely-packed points. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "38dd26d2-d9e5-4c25-86d9-cb0108f4ca64",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "m = nybb.explore()\n",
+    "nybb.sample_points(method='grid', tile='hex', size=10).explore(m=m, color='red')"
```

Comment:
This example seems to have some strange artifact:

![image](https://user-images.githubusercontent.com/1020496/223744333-7599faef-40dd-441a-b642-953400fe6f39.png)

Reply:
I'll have to take a look as to why this is happening... it should be covered completely, and was in earlier editions.

Reply:
Maybe worth checking if it is not an issue with folium, I noticed some weirdness lately.